plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveFileOperations.java

```diff
@@ -0,0 +1,189 @@
+import java.util.regex.Pattern;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.trino.filesystem.tracing.FileSystemAttributes.FILE_LOCATION;
+import static io.trino.plugin.hive.HiveQueryRunner.HIVE_CATALOG;
+import static io.trino.testing.MultisetAssertions.assertMultisetsEqual;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toCollection;
+
+@Execution(ExecutionMode.SAME_THREAD)
+public class TestHiveFileOperations
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected DistributedQueryRunner createQueryRunner()
+            throws Exception
+    {
+        Path metastoreDirectory = Files.createTempDirectory(HIVE_CATALOG);
+        closeAfterClass(() -> deleteRecursively(metastoreDirectory, ALLOW_INSECURE));
+
+        Map<String, String> hiveProperties = ImmutableMap.<String, String>builder()
+                .put("hive.file-status-cache-tables", "*")
+                .put("hive.metastore", "file")
+                .put("hive.metastore.catalog.dir", metastoreDirectory.toUri().toString())
+                .buildOrThrow();
+
+        return HiveQueryRunner.builder()
+                .setCoordinatorProperties(ImmutableMap.of("node-scheduler.include-coordinator", "false"))
+                .setHiveProperties(hiveProperties)
+                .setWorkerCount(1)
+                .build();
+    }
+
+    @Test
+    public void testFileStatusFileOperations()
+    {
+        assertQuerySucceeds("CALL system.flush_metadata_cache()");
+        assertUpdate("DROP TABLE IF EXISTS test_cache_file_operations");
+        assertUpdate("CREATE TABLE test_cache_file_operations(data varchar, key varchar) WITH (partitioned_by=ARRAY['key'], format='parquet')");
+        assertUpdate("INSERT INTO test_cache_file_operations VALUES ('1-abc', 'p1')", 1);
+        // FileSystem.listFiles, test_cache_file_operations is from listing partitions in FileHiveMetastore, so can be ignored
+        assertFileSystemAccesses(
+                "SELECT * FROM test_cache_file_operations",
+                ImmutableMultiset.<FileOperation>builder()
+                        .add(new FileOperation("Input.readTail", "key=p1/"))
+                        .add(new FileOperation("FileSystem.listFiles", "test_cache_file_operations"))
+                        .add(new FileOperation("FileSystem.listFiles", "key=p1"))
+                        .build());
+        assertFileSystemAccesses(
+                "SELECT * FROM test_cache_file_operations",
+                ImmutableMultiset.<FileOperation>builder()
+                        .add(new FileOperation("Input.readTail", "key=p1/"))
+                        .add(new FileOperation("FileSystem.listFiles", "test_cache_file_operations"))
+                        .build());
+
+        assertUpdate("INSERT INTO test_cache_file_operations VALUES ('2-xyz', 'p2')", 1);
```

Comment:
What if we insert into the same parttition ?

Reply:
Cases like that are already tested in `BaseCachingDirectoryListerTest`